### PR TITLE
Sre cleanup9

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
This PR moves the token assignment for unfinished types etc. to the BCL instead of the runtime. This allows the deletion of a lot of C code. There should be no change in functionality, except the temporary tokens.

File: mcs/class/corlib/System.Reflection.Emit/ConstructorOnTypeBuilderInst.cs [hunk 1]
```diff
@@ -134,6 +134,26 @@ internal override ParameterInfo[] GetParametersInternal ()
 			return res;
 		}
 
+		internal override Type[] GetParameterTypes () {
+			if (cb is ConstructorBuilder) {
```
Comment (marek-safar) [inline mcs/class/corlib/System.Reflection.Emit/ConstructorOnTypeBuilderInst.cs]: `as` would be better

File: mcs/class/corlib/System.Reflection.Emit/ILGenerator.cs [hunk 2]
```diff
@@ -755,7 +755,7 @@ private void Emit (OpCode opcode, MethodInfo method, int token)
 			// Might be a DynamicMethod with no declaring type
 			Type declaringType = method.DeclaringType;
 			if (declaringType != null) {
-				if (declaringType.Module == module)
+				if (declaringType.Module == module || method is MethodBuilder)
```
Comment (marek-safar) [inline mcs/class/corlib/System.Reflection.Emit/ILGenerator.cs]: Isn't `method is MethodBuilder` superior check to `declaringType.Module == module`

Comment (vargaz) [inline mcs/class/corlib/System.Reflection.Emit/ILGenerator.cs]: I guess the first case can happen without the second for already finished types.

**Comment by vargaz** [comment]
Any other comments ? @kumpera.

**Pull request merged**